infernyx/rules.py

```diff
@@ -297,9 +343,49 @@ def tag_results(suffix, job):
                 key_parts=['date', 'locale', 'country_code', 'os', 'browser', 'version', 'device', 'year',
                            'month', 'week'],
                 value_parts=['newtabs'],
-                table='newtab_stats_daily',
-            ),
-        },
+                table='newtab_stats_daily')
+        }
+    ),
+    InfernoRule(
+        name='blacklisted_impression_stats',
+        source_tags=['incoming:impression'],
+        day_range=1,
+        day_offset=1,
+        time_delta={'oclock': 4},
+        rule_cleanup=report_rule_stats,
+        map_input_stream=chunk_json_stream,
+        map_init_function=impression_stats_init,
+        blacklisted=True,
+        parts_preprocess=[clean_data, parse_date, parse_locale, parse_ip, filter_blacklist, parse_ua, parse_tiles],
+        geoip_file=GEOIP,
+        partitions=32,
+        sort_buffer_size='25%',
+        locale_whitelist={'ach', 'af', 'an', 'ar', 'as', 'ast', 'az', 'be', 'bg', 'bn-bd', 'bn-in', 'br', 'bs',
```

Comment:
Good catch, Nan.  I fixed that in the latest push

On Thu, May 28, 2015 at 10:32 AM, Nan Jiang <notifications@github.com>
wrote:

> In infernyx/rules.py
> <https://github.com/tspurway/infernyx/pull/20#discussion_r31238958>:
>
> > +    ),
> > +    InfernoRule(
> > +        name='blacklisted_impression_stats',
> > +        source_tags=['incoming:impression'],
> > +        day_range=1,
> > +        day_offset=1,
> > +        time_delta={'oclock': 4},
> > +        rule_cleanup=report_rule_stats,
> > +        map_input_stream=chunk_json_stream,
> > +        map_init_function=impression_stats_init,
> > +        blacklisted=True,
> > +        parts_preprocess=[clean_data, parse_date, parse_locale, parse_ip, filter_blacklist, parse_ua, parse_tiles],
> > +        geoip_file=GEOIP,
> > +        partitions=32,
> > +        sort_buffer_size='25%',
> > +        locale_whitelist={'ach', 'af', 'an', 'ar', 'as', 'ast', 'az', 'be', 'bg', 'bn-bd', 'bn-in', 'br', 'bs',
>
> There is another copy of the locale_whitelist in rule impression_stats,
> better take it out and reuse it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tspurway/infernyx/pull/20/files#r31238958>.
>